### PR TITLE
Apply clang-format

### PR DESCRIPTION
**Pull request by rakhimov**
The style configuration is derived from the existing source files.
However, there are some inconsistencies among the existing files.
This is fixed with clang-format.

### CODE REVIEW

File: .clang-format [hunk 1]
```diff
@@ -0,0 +1,29 @@
+BasedOnStyle: llvm
```
Comment (andrewprock) [inline .clang-format]: Should be based on google's style guide.

File: .clang-format [hunk 2]
```diff
@@ -0,0 +1,29 @@
...
+AccessModifierOffset: -4
+BreakBeforeBinaryOperators: None
+ColumnLimit: 80
+Standard: Cpp11
+IndentWidth: 4
+AlwaysBreakTemplateDeclarations: true
+BreakBeforeBraces: Custom
+BraceWrapping:
```
Comment (andrewprock) [inline .clang-format]: Should be almost always false.

File: src/programs/ps-lut/main.cpp [hunk 3]
```diff
@@ -1,40 +1,42 @@
...
+int main(int argc, char** argv)
 {
-    try 
+    try
     {
-        // set up the program options, handle the help case, and extract the values
-        po::options_description desc(
-            "create a lookup table for poker hands\n");
-        desc.add_options()
-            ("help,?",    "produce help message")
```
Comment (andrewprock) [inline src/programs/ps-lut/main.cpp]: This is the preferred format. Structured for readability.

File: src/programs/ps-lut/main.cpp [hunk 4]
```diff
@@ -56,44 +58,37 @@ int main (int argc, char ** argv)
...
+                                bit->rankColex() % eval.str() % eval.code();
                 }
                 else
                 {
                     bit->canonize(*pit);
                     if (pit->intersects(*bit))
                         continue;
                     PokerHandEvaluation eval = evaluator->evaluate(*pit, *bit);
-                    cout << boost::format("%s, %s: [%4d,%4d] -> %s [%9d]\n") 
-                        % pit->str()
```
Comment (andrewprock) [inline src/programs/ps-lut/main.cpp]: keep separate line per expression.

File: .clang-format [hunk 5]
```diff
@@ -0,0 +1,29 @@
...
+    AfterEnum: true
+    AfterStruct: true
+    AfterUnion: true
+    AfterExternBlock: true
+    BeforeCatch: true
+    BeforeElse: true
+    AfterNamespace: true
+    IndentBraces: false
+    SplitEmptyFunction: false
+AllowShortFunctionsOnASingleLine: Empty
```
Comment (andrewprock) [inline .clang-format]: Should be all.

File: .clang-format [hunk 6]
```diff
@@ -0,0 +1,29 @@
...
+DerivePointerAlignment: false
+PointerAlignment: Left
+SpacesBeforeTrailingComments: 2
+BreakConstructorInitializersBeforeComma: true
+BreakConstructorInitializers: BeforeColon
```
Comment (andrewprock) [inline .clang-format]: Should be BeforeComma

File: src/programs/ps-lut/main.cpp (hunk 4 above)
Comment (rakhimov) [reply src/programs/ps-lut/main.cpp]: I am not sure how to enforce this automatically.

File: .clang-format (hunk 1 above)
Comment (andrew-ebss) [reply .clang-format]: I prefer Google style myself, but this code was written a while ago.

If you just set it as Google style, with longer lines, and don't reformat horizontal alignment, I think that would be good.

File: src/programs/ps-lut/main.cpp (hunk 4 above)
Comment (andrew-ebss) [reply src/programs/ps-lut/main.cpp]: You may have to just not reformat this file.

Comment (rakhimov) [reply src/programs/ps-lut/main.cpp]: Yes, I disabled clang-format for this specific pieces of code,
but it somehow not showing up on this diff.

File: .clang-format (hunk 1 above)
Comment (rakhimov) [reply .clang-format]: > don't reformat horizontal alignment

Not sure if this is possible to tell to clang-format.

File: src/programs/ps-lut/main.cpp (hunk 3 above)
Comment (rakhimov) [reply src/programs/ps-lut/main.cpp]: This is reformatted manually but not showing up on this diff.

**Pull request merged**